src/sobol_sensitivity.jl

```diff
@@ -342,10 +342,11 @@ function gsa_sobol_all_y_analysis(method, all_y::AbstractArray{T}, d, n, Ei_esti
         nboot > 1 ? reshape(ST_CI, size_...) : nothing)
 end
 
-function gsa(f, method::Sobol, p_range::AbstractVector; samples, kwargs...)
+function gsa(f, method::Sobol, p_range::AbstractVector; samples, rng::AbstractRNG = Random.default_rng(), kwargs...)
     AB = QuasiMonteCarlo.generate_design_matrices(samples, [i[1] for i in p_range],
         [i[2] for i in p_range],
         QuasiMonteCarlo.SobolSample(),
+        QuasiMonteCarlo.OwenScramble(; base=2, pad=ceil(Int, log2(samples)), rng),
```

Comment:
The API for `generate_design_matrices` is surprisingly confusing. But I think the following should work:
```suggestion
        QuasiMonteCarlo.SobolSample(; R=QuasiMonteCarlo.OwenScramble(; base=2, pad=ceil(Int, log2(samples)), rng)),
```